variableComments

FullStackAbbs opened the pull request:
Added comments to the variables under findKnife

FullStackAbbs committed:
variableComments

Darian-the-Yeti commented on js/interviewQs/knives.js:
```diff
@@ -57,9 +57,10 @@ class Solution {
-    const tolerance = .15
-    const upBound = psiSliceSelected + tolerance
-    const lowBound = psiSliceSelected - tolerance
+    // psiSliceSelected (pounds per square inch) refers to the amount of pressure needed for a succesful cut
+    const tolerance = .15 // .15 psi range in order to find the correct knife
+    const upBound = psiSliceSelected + tolerance // the pressure need to make a cut while with tolerance range (high)
+    const lowBound = psiSliceSelected - tolerance // the pressure need to make a cut while with tolerance range (low)
```
These are good!

Darian-the-Yeti reviewed (approved):
I will approve this because you met the requirements for this PR. Congrats again on your first PR! 

Next I want you to make some changes: 
 - Make a PR that creates a `knife` class and a `food` class.
    - The knife class should have the properties you've mentioned so far, but it should also have a `cut` method that accepts some food, and [throws](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw) an error if it can't cut it.
        - It can't cut if the knife doesn't match the psiSlice logic you've made or if the food `isMoldy`
        - It should throw an error with the reason for why it's not able to cut the food
    - The `food` class should have the properties you've mentioned here but also a `spoilAge` property and a `currentAge` property.
        - The user should be able to set currentAge AFTER making the object that's made from the `food` class. So maybe make a setter method
    - The `food` class should also have method (`isMoldy()`) that checks whether or not the food is moldy and returns true or false.

FullStackAbbs committed:
update

still figuring out how to access a obj property within in class which I believe is needed for the cut method in the Knife class

Darian-the-Yeti commented on js/interviewQs/knives.js:
```diff
@@ -30,21 +30,59 @@ const knives = {
+  cut(psi){
+    if(food.psiSlice < this.psi ){
+      return 'This cannot cut '
+
```
```suggestion
```

Darian-the-Yeti commented on js/interviewQs/knives.js:
```diff
@@ -30,21 +30,59 @@ const knives = {
+        psiSlice: .56,
+        spoilAge: 10,
+        currentAge: 2 
     
```
Extra new line not needed
```suggestion
```

Darian-the-Yeti commented on js/interviewQs/knives.js:
```diff
@@ -30,21 +30,59 @@ const knives = {
+    this.location = location
+  }
+  cut(psi){
+    if(food.psiSlice < this.psi ){
```
where does this food come from?

Darian-the-Yeti commented on js/interviewQs/knives.js:
```diff
@@ -30,21 +30,59 @@ const knives = {
+}
+
+let knife4 = new Knife('Boning Knife',3,7.5,1.24,'C1')
+console.log(knife4.cut())
```
have you tried running this? there is an issue here, let's see if you can find it  :)

Darian-the-Yeti commented on js/interviewQs/knives.js:
```diff
@@ -30,21 +30,59 @@ const knives = {
+  }
+  isMoldy(spoilAge,currentAge){
+    this.currentAge > this.spoilAge ? true : false
+}}
```
This formatting makes things seem a bit confusing, can you put these on a different line?

FullStackAbbs replied:
> This formatting makes things seem a bit confusing, can you put these on a different line?

this was a ternary operator I decided to use due to the output possibilities being limited. I changed this is the next pull request.

Darian-the-Yeti commented on js/interviewQs/knives.js:
```diff
@@ -30,21 +30,59 @@ const knives = {
+    this.spoilAge = spoilAge
+    this.currentAge =currentAge
+  }
+  isMoldy(spoilAge,currentAge){
```
what are we missing in this function?

FullStackAbbs replied:
possibly the setter function to change the age, possible currentAge plus one for the day increment

Darian-the-Yeti reviewed (commented):
We will talk about this now

FullStackAbbs commented on js/interviewQs/knives.js:
```diff
@@ -30,21 +30,59 @@ const knives = {
+    this.psi = psi
+    this.location = location
+  }
+  cut(psi){
```
Change the parameters from psi to accept an input of any misc. food obj insert

Function Call Example: ButcherKnife.cut('Onions').....

FullStackAbbs committed:
ok

FullStackAbbs committed:
chop() method added

FullStackAbbs committed:
Update .DS_Store
